dpdk/src/mem.rs

```diff
@@ -399,7 +402,74 @@ pub struct Mbuf {
 impl Drop for Mbuf {
     fn drop(&mut self) {
         unsafe {
-            dpdk_sys::rte_pktmbuf_free_w(self.raw.as_ptr());
+            dpdk_sys::rte_pktmbuf_free(self.raw.as_ptr());
+        }
+    }
+}
+
+impl AsRef<[u8]> for Mbuf {
+    fn as_ref(&self) -> &[u8] {
+        self.raw_data()
+    }
+}
+
+impl AsMut<[u8]> for Mbuf {
+    fn as_mut(&mut self) -> &mut [u8] {
+        self.raw_data_mut()
+    }
+}
+
+impl Headroom for Mbuf {
+    fn headroom(&self) -> u16 {
+        unsafe { rte_pktmbuf_headroom(self.raw.as_ptr()) }
```

Comment:
How does this work, how does it know how much headroom, can it fail?

Reply:
[DPDK is tracking the headroom](https://doc.dpdk.org/guides/prog_guide/mbuf_lib.html#figure-mbuf1)